Tools/CISupport/ews-build/is_failing.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2020, 2021 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import requests
+import sys
+
+
+class ResultsDatabase(object):
+    URL = 'https://results.webkit.org'
+    SUITE = 'layout-tests'
+    THRESHOLD = 10
+
+    @classmethod
+    def summary(cls, test, commit=None, configuration=None):
+        params = dict(**(configuration or dict()))
+        if commit:
+            params['ref'] = commit
+        response = requests.get(
+            '{}/api/results-summary/{}/{}'.format(cls.URL, cls.SUITE, test),
+            params=params,
+        )
+        if response.status_code != 200:
+            return None
+        return response.json()
+
+    @classmethod
+    def expected(cls, test, type=None, commit=None, configuration=None, log=False):
+        data = cls.summary(test, commit=commit, configuration=configuration)
```

Comment:
what happens if the passed commit sha /identifier doesn't yield any data on results db, does results db produce results based on nearby commits?

Reply:
Yes, even if the tests did give a result on the specific commit, we still take into account commits around it. There is an algorithm which basically applies weights to each result based on how far it is from the specified commit, closer being given more weight.